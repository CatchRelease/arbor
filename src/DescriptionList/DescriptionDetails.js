import Box from '../Box';

const DescriptionDetails = Box.withComponent('dd');

DescriptionDetails.defaultProps = {
  ...Box.defaultProps,
  color: 'text.default',
  fontSize: 'size4',
  lineHeight: 'small',
  mb: 'regular'
};

export default DescriptionDetails;
