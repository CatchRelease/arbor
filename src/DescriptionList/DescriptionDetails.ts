import Box from '../Box';

const DescriptionDetails = Box.withComponent('dd');

DescriptionDetails.defaultProps = {
  color: 'text.default',
  fontSize: 'size4',
  lineHeight: 'small',
  mb: 'regular'
};

export default DescriptionDetails;
