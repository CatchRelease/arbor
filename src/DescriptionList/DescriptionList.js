import Box from '../Box';

const DescriptionList = Box.withComponent('dl');

export default DescriptionList;
