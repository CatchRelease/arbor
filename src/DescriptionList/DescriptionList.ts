import styled from '@emotion/styled';
import { css, Theme } from '@emotion/react';

import Box from '../Box';
import { PolyComponent } from '../polyComponent';

const inlineStyles = ({
  inline,
  theme
}: {
  inline?: boolean;
  theme: Theme;
}) => {
  if (!inline) {
    return '';
  }

  return css`
    white-space: nowrap;
    max-width: 100%;
    overflow: hidden;
    text-overflow: ellipsis;

    dd {
      display: inline;
      margin-left: ${theme.space.smaller};
    }

    dd:after {
      display: block;
      content: '';
    }

    dt {
      display: inline-block;
      margin-bottom: ${theme.space.small};
    }
  `;
};

type Props = {
  inline?: boolean;
};

const DescriptionList: PolyComponent<'dl', Props> = styled(
  Box.withComponent('dl')
)<Props>`
  ${inlineStyles};
`;

export default DescriptionList;
