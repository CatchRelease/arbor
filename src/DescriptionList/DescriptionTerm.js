import Box from '../Box';

const DescriptionTerm = Box.withComponent('dt');

DescriptionTerm.defaultProps = {
  ...Box.defaultProps,
  color: 'text.muted',
  fontSize: 'size3',
  lineHeight: 'small',
  mb: 'smallest'
};

export default DescriptionTerm;
