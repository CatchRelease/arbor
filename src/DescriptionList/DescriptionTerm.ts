import Box from '../Box';

const DescriptionTerm = Box.withComponent('dt');

DescriptionTerm.defaultProps = {
  color: 'text.muted',
  fontSize: 'size3',
  lineHeight: 'small',
  mb: 'smallest'
};

export default DescriptionTerm;
