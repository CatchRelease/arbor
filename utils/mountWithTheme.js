import { mount } from 'enzyme';
import { ThemeProvider } from '@emotion/react';
import theme from '../src/theme';

export default (Component, mountOptions = {}) => {
  const wrappedComponent = (
    <ThemeProvider theme={theme}>{Component}</ThemeProvider>
  );

  const mounted = mount(wrappedComponent, mountOptions);

  return mounted.find(Component.type);
};
