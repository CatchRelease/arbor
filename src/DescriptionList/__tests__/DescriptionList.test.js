import mountWithTheme from '../../../utils/mountWithTheme';

import DescriptionDetails from '../DescriptionDetails';
import DescriptionList from '../DescriptionList';
import DescriptionTerm from '../DescriptionTerm';

describe('<DescriptionList />', () => {
  it('renders the description list properly', () => {
    const descriptionList = mountWithTheme(
      <DescriptionList>
        <DescriptionTerm>Term 1</DescriptionTerm>
        <DescriptionDetails>Term 1 Details</DescriptionDetails>
        <DescriptionTerm>Term 2</DescriptionTerm>
        <DescriptionDetails>Term 2 Details</DescriptionDetails>
        <DescriptionTerm>Term 3</DescriptionTerm>
        <DescriptionDetails>Term 3 Details</DescriptionDetails>
      </DescriptionList>
    );

    expect(descriptionList).toMatchSnapshot();
  });
});
