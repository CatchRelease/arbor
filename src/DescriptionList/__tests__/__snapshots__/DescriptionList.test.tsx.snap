// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DescriptionList /> renders the description list properly 1`] = `
.emotion-0 {
  font-family: Motiva Sans;
  box-sizing: border-box;
}

.emotion-1 {
  font-family: Motiva Sans;
  color: #5B5D61;
  font-size: 0.8125rem;
  line-height: 1.25;
  margin-bottom: 0.25rem;
  box-sizing: border-box;
}

.emotion-2 {
  font-family: Motiva Sans;
  color: #25272D;
  font-size: 0.875rem;
  line-height: 1.25;
  margin-bottom: 1rem;
  box-sizing: border-box;
}

<Styled(dl)>
  <dl
    className="emotion-0"
  >
    <Styled(dt)
      color="text.muted"
      fontSize="size3"
      lineHeight="small"
      mb="smallest"
    >
      <dt
        className="emotion-1"
        color="text.muted"
        fontSize="size3"
      >
        Term 1
      </dt>
    </Styled(dt)>
    <Styled(dd)
      color="text.default"
      fontSize="size4"
      lineHeight="small"
      mb="regular"
    >
      <dd
        className="emotion-2"
        color="text.default"
        fontSize="size4"
      >
        Term 1 Details
      </dd>
    </Styled(dd)>
    <Styled(dt)
      color="text.muted"
      fontSize="size3"
      lineHeight="small"
      mb="smallest"
    >
      <dt
        className="emotion-1"
        color="text.muted"
        fontSize="size3"
      >
        Term 2
      </dt>
    </Styled(dt)>
    <Styled(dd)
      color="text.default"
      fontSize="size4"
      lineHeight="small"
      mb="regular"
    >
      <dd
        className="emotion-2"
        color="text.default"
        fontSize="size4"
      >
        Term 2 Details
      </dd>
    </Styled(dd)>
    <Styled(dt)
      color="text.muted"
      fontSize="size3"
      lineHeight="small"
      mb="smallest"
    >
      <dt
        className="emotion-1"
        color="text.muted"
        fontSize="size3"
      >
        Term 3
      </dt>
    </Styled(dt)>
    <Styled(dd)
      color="text.default"
      fontSize="size4"
      lineHeight="small"
      mb="regular"
    >
      <dd
        className="emotion-2"
        color="text.default"
        fontSize="size4"
      >
        Term 3 Details
      </dd>
    </Styled(dd)>
  </dl>
</Styled(dl)>
`;

exports[`<DescriptionList /> supports polymorphic as prop 1`] = `
.emotion-0 {
  font-family: Motiva Sans;
  box-sizing: border-box;
}

.emotion-1 {
  font-family: Motiva Sans;
  box-sizing: border-box;
  display: grid;
  grid-template-columns: 1fr;
  font-family: Motiva Sans;
  box-sizing: border-box;
}

.emotion-2 {
  font-family: Motiva Sans;
  color: #5B5D61;
  font-size: 0.8125rem;
  line-height: 1.25;
  margin-bottom: 0.25rem;
  box-sizing: border-box;
}

.emotion-3 {
  font-family: Motiva Sans;
  color: #25272D;
  font-size: 0.875rem;
  line-height: 1.25;
  margin-bottom: 1rem;
  box-sizing: border-box;
}

<Styled(dl)
  as={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "__emotion_base": "div",
      "__emotion_forwardProp": undefined,
      "__emotion_real": [Circular],
      "__emotion_styles": Array [
        "
  ",
        [Function],
        ";

  ",
        [Function],
        ";
  ",
        [Function],
        ";
  ",
        [Function],
        ";
  ",
        [Function],
        ";
  ",
        [Function],
        ";
  ",
        [Function],
        ";
  ",
        [Function],
        ";
  ",
        [Function],
        ";
  ",
        [Function],
        ";
  ",
        [Function],
        ";
  ",
        [Function],
        ";
  ",
        [Function],
        ";
  ",
        [Function],
        ";
",
        "
  box-sizing: border-box;
  ",
        [Function],
        ";
  ",
        [Function],
        ";
  ",
        [Function],
        ";
  ",
        [Function],
        ";
  ",
        [Function],
        ";
  ",
        [Function],
        ";
  ",
        [Function],
        ";
  ",
        [Function],
        ";
  ",
        [Function],
        ";
",
        "
  ",
        [Function],
        ";
  ",
        [Function],
        ";
  ",
        [Function],
        ";
  ",
        [Function],
        ";
  ",
        [Function],
        ";
",
      ],
      "defaultProps": Object {
        "display": "grid",
      },
      "render": [Function],
      "withComponent": [Function],
    }
  }
  gridTemplateColumns="1fr"
>
  <Styled(div)
    className="emotion-0"
    display="grid"
    gridTemplateColumns="1fr"
  >
    <div
      className="emotion-1"
      display="grid"
    >
      <Styled(dt)
        color="text.muted"
        fontSize="size3"
        lineHeight="small"
        mb="smallest"
      >
        <dt
          className="emotion-2"
          color="text.muted"
          fontSize="size3"
        >
          Term 1
        </dt>
      </Styled(dt)>
      <Styled(dd)
        color="text.default"
        fontSize="size4"
        lineHeight="small"
        mb="regular"
      >
        <dd
          className="emotion-3"
          color="text.default"
          fontSize="size4"
        >
          Term 1 Details
        </dd>
      </Styled(dd)>
    </div>
  </Styled(div)>
</Styled(dl)>
`;
